io/iouring-wrapper.cpp

```diff
@@ -76,10 +76,14 @@ class iouringEngine : public MasterEventEngine, public CascadingEventEngine, pub
     }
 
     int init() {
-        rlimit resource_limit{.rlim_cur = RLIM_INFINITY, .rlim_max = RLIM_INFINITY};
-        if (setrlimit(RLIMIT_MEMLOCK, &resource_limit) != 0) {
-            LOG_ERROR_RETURN(0, -1, "iouring: failed to set resource limit. Use command `ulimit -l unlimited`, or change to root");
+        int compare_result;
+        if (kernel_version_compare("5.11", compare_result) == 0 && compare_result <= 0) {
+            rlimit resource_limit{.rlim_cur = RLIM_INFINITY, .rlim_max = RLIM_INFINITY};
+            if (setrlimit(RLIMIT_MEMLOCK, &resource_limit) != 0)
+                LOG_ERROR_RETURN(0, -1, "iouring: failed to set resource limit. "
```

Comment:
is ERROR_RETURN absolutely needed?

Reply:
Because we need to set a value, and we have no idea what's the appropriate number, based on the ring size.

So a strong check makes things easier, and let users know in advance.